File: 卡码/代码随想录/JS/动态规划/01-基础题目/LeetCode_0096.js
//96. 不同的二叉搜索树
const numTrees = function(n) {
	const dp=new Array(n+1).fill(0);
	dp[0]=1;
	for(let i=1;i<n+1;i++){
		for(let j=1;j<=i;j++){
			dp[i]+=dp[j-1]*dp[i-j];
		}
	}
	return dp[n];
};

File: 卡码/代码随想录/Go/动态规划/01-基础题目/LeetCode_0096.go
//96. 不同的二叉搜索树
package main

func numTrees(n int) int {
	dp:=make([]int,n+1)
	for i:=0;i<n+1;i++ {
		dp[i]=0
	}
	dp[0]=1
	for i:=1;i<n+1;i++ {
		for j:=1;j<=i;j++ {
			dp[i]+=dp[j-1]*dp[i-j]
		}
	}
	return dp[n]
}
